firmware/shared/os/event_flags.h

```diff
@@ -0,0 +1,17 @@
+/// @author Matteo Tullo
+/// @date 2024-02-24
+
+#pragma once
+
+#include <cstdint>
+#include "shared/util/os.h"
+
+namespace shared::osDataType {
```

Comment:
Why can't this just be called shared::os? There shouldn't be a need to separate the os data types from other os code using namespace

Reply:
As per discussion, removed osDataType and converted all RTOS related classes and functions to use the `::os` namespace